### ISSUE TITLE
Add ExternalAudioTrackSource

### ISSUE THREAD

**Issue by TFleury** [microsoft/MixedReality-WebRTC#345]
For now, the only option to create a `LocalAudioTrack` is from a local device (and you can't select it).

I would want to have an `ExternalAudioTrackSource` like we already have the `ExternalVideoTrackSource`.

Ortherwise, is there another way to send audio frames directly ?

**Comment by djee-ms** [comment]
Nope, there's no other way, we need to add something like this if you need any other source.

**Comment by djee-ms** [comment]
I had a look at this and it proved much more complex than initially expected. The audio module does not follow the audio source interface pattern like the video one, and instead always grab audio frames for all tracks from the internal audio module, so it seems customizing is much more limited. I asked for help/feedback on https://groups.google.com/g/discuss-webrtc/c/3H4vwWgNuQk and I am waiting to see how this can be done, but basically for now I am a bit stuck.

**Comment by pgoorts** [comment]
Is there any news on this feature?

**Comment by alexdegroot** [comment]
@djee-ms any updates on this? Do you still agree that the best way to implement this would be a custom dispatch ADM?

**Comment by djee-ms** [comment]
Sounds like it. But I am not working anymore on this project, see Notice on [README.md](https://github.com/microsoft/MixedReality-WebRTC/blob/master/README.md).

**Comment by timothyghanim** [comment]
Any updates on this?

We're desperately looking for this feature. I want to be able to relay audio frames between two connections and so the only way to do that is through an audio frame request callback similar to the one in `ExternalVideoTrackSource`.

If there are no updates yet, how can I relay audio frames from a `RemoteAudioTrack` to a `LocalAudioTrack`?

Thanks in advance.

**Comment by alexdegroot** [comment]
@tghanem for one of my clients we've abandoned the MixedReality-WebRTC library. We tried to reach out using issues and the emails mentioned in the README.md. Basically 0 response. After careful consideration, we moved to https://github.com/sipsorcery-org/sipsorcery

**Comment by timothyghanim** [comment]
Thanks, @alexdegroot! I discovered **sipsorcery** while looking for a WebRTC library written in .NET but I thought a Microsoft maintained library would be a better option. Unfortunately, it seems not!
